In [1]:
import numpy as np
from mostlyai import MostlyAI

In [2]:
# initialize client
mostly = MostlyAI(api_key='mostly-adcd77a4ca7523519dc19b4ec9c9a7f6e0b2e37e16599b4703b03944dc824ed2', base_url='https://app.mostly.ai')

In [3]:
# fetch configuration via API
g = mostly.generators.get('1d9f85fd-277c-40d8-985f-02d3f39c948b')
config = g.config()
config

{'name': 'SK Patient Data',
 'description': None,
 'tables': [{'name': 'SK Patient Data',
   'sourceConnectorId': '2f880c16-adf9-4b89-85a6-29bb9316c741',
   'location': 'mostly-production-mostly-s3-bucket-mostly-app/connectors/2f880c16-adf9-4b89-85a6-29bb9316c741/SK Patient Data/cleaned_sk_data.csv',
   'data': None,
   'modelConfiguration': {'maxSampleSize': 6712,
    'batchSize': None,
    'modelSize': 'M',
    'maxTrainingTime': 10,
    'maxEpochs': 100,
    'maxSequenceWindow': 100,
    'enableFlexibleGeneration': False,
    'valueProtection': False,
    'rareCategoryReplacementMethod': 'CONSTANT',
    'compute': 'd5f093a6-208f-4b79-92c6-6d50d02d8c8c'},
   'textModelConfiguration': None,
   'primaryKey': None,
   'foreignKeys': None,
   'columns': [{'name': 'latitude', 'modelEncodingType': 'NUMERIC_BINNED'},
    {'name': 'longitude', 'modelEncodingType': 'NUMERIC_BINNED'},
    {'name': 'sex', 'modelEncodingType': 'NUMERIC_DISCRETE'},
    {'name': 'age', 'modelEncodingType': 'NUMERI

In [4]:
# probe the generator for some samples
mostly.probe('1d9f85fd-277c-40d8-985f-02d3f39c948b', size=10)

╭──────────────────────────────────────────────────────╮
│ HTTP 403: You have reached the limit of 5.0 credits. │
╰──────────────────────────────────────────────────────╯

APIStatusError: HTTP 403: You have reached the limit of 5.0 credits.

We need 20 synthetic data sets. Original data set size was 6712, so we need $6712 \times 20 = 134240$ samples.

In [6]:
# use generator to create a synthetic dataset
sd = mostly.generate('1d9f85fd-277c-40d8-985f-02d3f39c948b', size=134_240)
sd.data()

Created synthetic dataset ]8;id=162747;https://app.mostly.ai/d/synthetic-datasets/0e69c80f-8bc4-4653-92c8-740ace235996\0e69c80f-8bc4-4653-92c8-740ace235996]8;;\ with generator ]8;id=420694;https://app.mostly.ai/d/generators/1d9f85fd-277c-40d8-985f-02d3f39c948b\1d9f85fd-277c-40d8-985f-02d3f39c948b]8;;\

Started synthetic dataset generation

Output()

🎉 Your synthetic dataset is ready! Use it to consume the generated data. Share it so others can do the same.

latitude  longitude  sex  age  state
0     -1.182548   1.997387    0    2      0
1       0.76447  -1.224311    1    4      0
2      0.795372  -0.777893    0    4      0
3     -1.708918   1.049877    0    2      0
4       0.88762  -0.807637    0    4      0
...         ...        ...  ...  ...    ...
67115  0.809655  -0.783382    0    4      0
67116 -0.976759   1.070693    0    2      0
67117 -0.463864  -0.190423    1    2      0
67118 -0.345393  -0.290523    1    5      0
67119  0.821947   -0.80782    0    4      0

[134240 rows x 5 columns]

Split the generated synthetic data into 20 data sets.

In [9]:
split_sd = np.array_split(sd.data(), 20)

C:\ProgramData\anaconda3\Lib\site-packages\numpy\core\fromnumeric.py:59: FutureWarning: 'DataFrame.swapaxes' is deprecated and will be removed in a future version. Please use 'DataFrame.transpose' instead.
  return bound(*args, **kwds)


In [12]:
for i, sX in enumerate(split_sd):
    sX.to_csv("../Data/SK/Synthetic Datasets/mostly_" + str(i) + ".csv", index=False)